### ISSUE TITLE
Is ARM support still broken?

### ISSUE THREAD

**Issue by gschmottlach-xse** [jmckaskill/luaffi#50]
Just wanted to see where this project was left off. Is the ARM support still broken? I'd like to use this on an i.MX6 class device (ARM Cortex A9). Is there any reason this would _not_ work? How extensively has it been tested (e.g. what is known to be broken)?

Any feedback would be appreciated.

**Comment by jmckaskill** [comment]
Yes it is. I refactored the API between the main code and the platform specific code awhile back and didn't have any ARM hardware at the time so left it with the previous API. Unfortunately I've been busy with other projects since then and so haven't had a chance to get back to it.

You might be able to get functions working without complex, long double or callbacks by just updating it to compile.
